pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -17,22 +17,55 @@ limitations under the License.
 package util
 
 import (
+	"sync"
+
 	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-// GetNamespacesFromPodAffinityTerm returns a set of names
-// according to the namespaces indicated in podAffinityTerm.
-// If namespaces is empty it considers the given pod's namespace.
+// NamespaceLister helps list Namespaces.
+type namespaceLister interface {
+	// List lists all Namespaces in the indexer.
+	List(selector labels.Selector) (ret []*v1.Namespace, err error)
+}
+
+var (
+	once     sync.Once
+	nsLister namespaceLister
+)
+
+func InitNamespaceLister(namespaceListerLister namespaceLister) {
+	once.Do(func() {
+		nsLister = namespaceListerLister
+	})
+}
+
 func GetNamespacesFromPodAffinityTerm(pod *v1.Pod, podAffinityTerm *v1.PodAffinityTerm) sets.String {
-	names := sets.String{}
-	if len(podAffinityTerm.Namespaces) == 0 {
-		names.Insert(pod.Namespace)
+	namespaces := sets.String{}
+	if podAffinityTerm.NamespaceSelector != nil {
```

Comment:
When the selector is not provided (is nil), shouldn't we treat it like the case that it is provided and is empty?

Reply:
I don't think so, consider a user didn't set iterm.Namespaces, means they depends on the pod's namespace(previous logic), now, we added the namespace selector, if we treat it like the case it is empty, the namespaces will be a empty list, this will break the backward compatability.

Reply:
That's a good point, but shouldn't we add the pod's namespace to the `namespaces` when the NamespaceSelector is nil.?

Reply:
Oh, nevermind. The current code does that. So, it is all fine. Please address my other comment and the rest LGTM.